doc/lua_api.md

```diff
@@ -8898,7 +8898,10 @@ Player properties need to be saved manually.
 Entity definition
 -----------------
 
-Used by `minetest.register_entity`.
+Used by `minetest.register_entity`.  
+The entity definition table becomes a metatable of a newly created per-entity
+table, meaning its subtables (e.g. `initial_properties`) will be shared between
+all instances of an entity
```

Comment:
```suggestion
The entity definition table becomes a metatable of a newly created per-entity
luaentity table, meaning its fields (e.g. `initial_properties`) will be shared
between all instances of an entity.
```
The fields are not just tables.

Also, mentioning subtables makes it sound to me like you had metatable semantics *in* the subtables, which is not the case. If you override a field in `luaentity.initial_properties`, you modify the prototype.

Btw., FYI, but probably not worth mentioning, being set as a metatable isn't enough make the index magic work. What also happens is that the `__index` field is set (see `prototype.__index = prototype` in `core.register_entity` in builtin).

Reply:
With your suggestion it sounds like every field is shared, which is not the case. If I have `foo = 5` as a custom field, that won't be shared. If I have `bar = { foo = 5 }` that will

Reply:
Relevant code: 

https://github.com/minetest/minetest/blob/1222750c50c3077d5793675bfdc337b66ed24313/src/script/cpp_api/s_entity.cpp#L44-L70

`core.registered_entities[name]` is used as a metatable of the new entity table. Perhaps it's the most technically correct way to describe it as

```suggestion
The entity definition table becomes a metatable of a newly created per-entity
luaentity table, roughly meaning its fields (e.g. `initial_properties`) will be
provided as default values.
```

Reply:
It is correct but can be confusing to people less familar with Lua. 

in the following example you won't have two copies of the string:
```lua
local a = "Minetest"
local b = a
```
it's just that strings are immutable so there's no way to mess it up

Reply:
>(or rather: everything except tables) are immutable

Functions (via `setfenv`), userdata, and coroutine threads are also mutable, so it's not just tables.

But I can see how my suggestion can cause confusion.

Krock's suggestion sounds good to me.

Reply:
> Krock's suggestion sounds good to me.

It emphasizes the "sharing" aspect even less, however.

Reply:
So do we want to change this? I like the initial suggestion from @Desour.

Reply:
I don't know, I personally find it confusing, but if two core devs agree who am I to say otherwise

Reply:
You can also make it longer and more explicit, e.g.:
```suggestion
The entity definition table becomes a metatable of a newly created per-entity
luaentity table, roughly meaning its fields (e.g. `initial_properties`) will be
provided as default values (but *not* deep copied).
```

If the modder knows lua well enough, they'll probably understand either way, because they know how metatables and values work.

Reply:
Went for the first one, simpler